modules/ximgproc/samples/slic.cpp

```diff
@@ -0,0 +1,138 @@
+#include <opencv2/imgproc.hpp>
+#include <opencv2/highgui.hpp>
+#include <opencv2/imgcodecs.hpp>
+#include <opencv2/core/utility.hpp>
+
+#include <opencv2/ximgproc.hpp>
+
+#include <ctype.h>
+#include <stdio.h>
+#include <iostream>
+
+using namespace cv;
+using namespace cv::ximgproc;
+using namespace std;
+
+static const char* window_name = "SLIC Superpixels";
+
+static const char* keys =
+    "{h help      | | help menu}"
+    "{c camera    |0| camera id}"
+    "{i image     | | image file}"
+    "{a algorithm |1| SLIC(0),SLICO(1)}"
+    ;
+
+int main(int argc, char** argv)
+{
+    CommandLineParser cmd(argc,argv,keys);
+    if (cmd.has("help")) {
+        cmd.about("This program demonstrates SLIC superpixels using OpenCV class SuperpixelSLIC.\n"
+            "If no image file is supplied, try to open a webcam.\n"
+            "Use [space] to toggle output mode, ['q' or 'Q' or 'esc'] to exit.\n");
+        cmd.printMessage();
+        return 0;
+    }
+    int capture = cmd.get<int>("camera");
```

Comment:
@StevenPuttemans , actually none of the arguments are mandatory. if you pass it a filename, it will happily work on that, else it will try to open the webcam. (the algorithm param is only there, to show the names)